### PR TITLE
Simplify build steps example

### PR DESCRIPTION
**Pull request by PatKamin**
### Description

Less steps in a build example in README:
- do not mix usage of make and cmake, cmake handles all that is needed
- cmake handles build dir creation

Preview: https://github.com/PatKamin/unified-memory-framework/tree/simplify-build-steps?tab=readme-ov-file#linux

### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [x] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

**Review by kswiecicki — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -47,22 +47,21 @@ For Level Zero memory provider tests:
 ### Linux
 
 Executable and binaries will be in **build/bin**
```
Comment (lukaszstolarczuk) [inline README.md:49]: pls add a full stop at the end of this line - it's all displayed as one sentence and it reads weird

// preview: https://github.com/PatKamin/unified-memory-framework/tree/simplify-build-steps?tab=readme-ov-file#linux

File: README.md [hunk 2]
```diff
@@ -47,22 +47,21 @@ For Level Zero memory provider tests:
...
-$ cmake {path_to_source_dir}
-$ make
+$ cmake -B build -DCMAKE_BUILD_TYPE={build_config}
+$ cmake --build build -j $(nproc)
 ```
 
 ### Windows
 
 Generating Visual Studio Project. EXE and binaries will be in **build/bin/{build_config}**
+`build_config` can be either `Debug` or `Release`.
```
Comment (lukaszstolarczuk) [inline README.md:60]: same here; perhaps add also "the":

> The `build_config`...

File: README.md (hunk 1 above)
Comment (PatKamin) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (PatKamin) [reply README.md]: Done

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**